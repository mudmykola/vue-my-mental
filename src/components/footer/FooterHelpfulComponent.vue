<script setup>
import { ref } from "vue";

const helpfulTitle = ref("Корисна інформація");

const helpfulMenu = ref([
	{
		link: "tel:0800505201",
		text: "0 800 505 201 - Гаряча лінія МОЗ",
	},
	{
		link: "tel:0800602019",
		text: "0 800 60 20 19 - Контакт-центр МОЗ",
	},
	{
		link: "moz.gov.ua",
		text: "moz.gov.ua",
	},
	{
		text:
			"7333 - Цілодобова гаряча лінія з попередження суїцидів та психологічної підтримки Lifeline Ukraine\n" +
			"для ветеранів",
	},
	{
		link: "tel:0800501701",
		text: "0–800–501–701 - Всеукраїнський телефон довіри",
	},
]);
</script>

<template>
  <div class="footer-helpful">
    <div class="footer-helpful__inner flex flex-col gap-6">
      <h2 class="text-[16px] font-bold">{{ helpfulTitle }}</h2>
      <ul class="flex flex-col gap-6">
        <li
          v-for="(item, index) in helpfulMenu"
          :key="index"
          class="flex flex-col gap-6"
        >
          <a
            :href="item.link"
            class="footer-helpful__link text-[14px] text-primary font-normal font-sans"
            >{{ item.text }}</a
          >
        </li>
      </ul>
    </div>
  </div>
</template>