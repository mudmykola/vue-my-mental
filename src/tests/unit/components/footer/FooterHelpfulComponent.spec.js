import FooterHelpfulComponent from "@/components/footer/FooterHelpfulComponent.vue";
import { mount } from "@vue/test-utils";

describe("FooterHelpfulComponent.vue", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(FooterHelpfulComponent);
	});

	it("renders correctly", () => {
		expect(wrapper.element).toMatchSnapshot();
	});

	it("displays the correct title", () => {
		const title = wrapper.find("h2");
		expect(title.text()).toBe("Корисна інформація");
	});

	it("displays the correct number of menu items", () => {
		const menuItems = wrapper.findAll("li");
		expect(menuItems.length).toBe(5);
	});

	it("displays the correct menu item texts", () => {
		const menuItems = wrapper.findAll("a");
		const expectedTexts = [
			"0 800 505 201 - Гаряча лінія МОЗ",
			"0 800 60 20 19 - Контакт-центр МОЗ",
			"moz.gov.ua",
			"7333 - Цілодобова гаряча лінія з попередження суїцидів та психологічної підтримки Lifeline Ukraine для ветеранів",
			"0–800–501–701 - Всеукраїнський телефон довіри",
		];

		menuItems.forEach((item, index) => {
			expect(item.text().replace(/\n\s*/g, " ")).toBe(expectedTexts[index]);
		});
	});

	it("displays the correct menu item links", () => {
		const menuItems = wrapper.findAll("a");
		const expectedLinks = [
			"tel:0800505201",
			"tel:0800602019",
			"moz.gov.ua",
			undefined,
			"tel:0800501701",
		];

		menuItems.forEach((item, index) => {
			expect(item.attributes("href")).toBe(expectedLinks[index]);
		});
	});
});