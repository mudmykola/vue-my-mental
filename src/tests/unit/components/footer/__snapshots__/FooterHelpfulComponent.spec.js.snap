// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterHelpfulComponent.vue renders correctly 1`] = `
<div
  class="footer-helpful"
>
  <div
    class="footer-helpful__inner flex flex-col gap-6"
  >
    <h2
      class="text-[16px] font-bold"
    >
      Корисна інформація
    </h2>
    <ul
      class="flex flex-col gap-6"
    >
      
      <li
        class="flex flex-col gap-6"
      >
        <a
          class="footer-helpful__link text-[14px] text-primary font-normal font-sans"
          href="tel:0800505201"
        >
          0 800 505 201 - Гаряча лінія МОЗ
        </a>
      </li>
      <li
        class="flex flex-col gap-6"
      >
        <a
          class="footer-helpful__link text-[14px] text-primary font-normal font-sans"
          href="tel:0800602019"
        >
          0 800 60 20 19 - Контакт-центр МОЗ
        </a>
      </li>
      <li
        class="flex flex-col gap-6"
      >
        <a
          class="footer-helpful__link text-[14px] text-primary font-normal font-sans"
          href="moz.gov.ua"
        >
          moz.gov.ua
        </a>
      </li>
      <li
        class="flex flex-col gap-6"
      >
        <a
          class="footer-helpful__link text-[14px] text-primary font-normal font-sans"
        >
          7333 - Цілодобова гаряча лінія з попередження суїцидів та психологічної підтримки Lifeline Ukraine
для ветеранів
        </a>
      </li>
      <li
        class="flex flex-col gap-6"
      >
        <a
          class="footer-helpful__link text-[14px] text-primary font-normal font-sans"
          href="tel:0800501701"
        >
          0–800–501–701 - Всеукраїнський телефон довіри
        </a>
      </li>
      
    </ul>
  </div>
</div>
`;
